src/main/java/org/opensearch/jobscheduler/utils/JobDetailsService.java

```diff
@@ -86,6 +132,129 @@ void updateIndexToJobDetails(String documentId, JobDetails jobDetails) {
         // Register new JobDetails entry
         indexToJobDetails.put(documentId, jobDetails);
         updateIndicesToListen(jobDetails.getJobIndex());
+        updateIndexToJobProviders(jobDetails);
+    }
+
+    /**
+     * Creates a proxy ScheduledJobParser that triggers an extension's jobParameter action
+     *
+     * @param extensionUniqueId the extension to trigger the job parameter action
+     * @param extensionJobParameterAction the job parameter action name
+     */
+    private ScheduledJobParser createProxyScheduledJobParser(String extensionUniqueId, String extensionJobParameterAction) {
+        return new ScheduledJobParser() {
+
+            @Override
+            public ScheduledJobParameter parse(XContentParser xContentParser, String id, JobDocVersion jobDocVersion) throws IOException {
+
+                logger.info("Sending ScheduledJobParameter parse request to extension : " + extensionUniqueId);
+
+                final ExtensionJobParameter[] extensionJobParameterHolder = new ExtensionJobParameter[1];
+                CompletableFuture<ExtensionJobParameter[]> inProgressFuture = new CompletableFuture<>();
+
+                // TODO : Replace the placeholder with the provided access token from the inital job detials request
+
+                // Prepare JobParameterRequest
+                JobParameterRequest jobParamRequest = new JobParameterRequest("placeholder", xContentParser, id, jobDocVersion);
+
+                // Invoke extension job parameter action and return ScheduledJobParameter
+                client.execute(
+                    ExtensionProxyAction.INSTANCE,
+                    new ExtensionJobActionRequest<JobParameterRequest>(extensionJobParameterAction, jobParamRequest),
+                    ActionListener.wrap(response -> {
+
+                        // Extract response bytes and generate the parsed job parameter
+                        JobParameterResponse jobParameterResponse = new JobParameterResponse(response.getResponseBytes());
+                        extensionJobParameterHolder[0] = jobParameterResponse.getJobParameter();
+                        inProgressFuture.complete(extensionJobParameterHolder);
+
+                    }, exception -> {
+
+                        logger.error("Could not parse job parameter", exception);
+                        inProgressFuture.completeExceptionally(exception);
+
+                    })
+                );
+
+                // Stall execution until request completes or times out
+                try {
+                    inProgressFuture.orTimeout(JobDetailsService.TIME_OUT_FOR_REQUEST, TimeUnit.SECONDS).join();
+                } catch (CompletionException e) {
+                    if (e.getCause() instanceof TimeoutException) {
+                        logger.info("Request timed out with an exception ", e);
```

Comment:
the way this is handled, this will just log this and not throw an exception.  You may not want "else" below this.